doubles/allowance.py

```diff
@@ -119,6 +120,17 @@ def with_args(self, *args, **kwargs):
         self.verify_arguments()
         return self
 
+    def with_args_that_match(self, matching_function):
```

Comment:
I agree about the name not being great.  I'm not a huge fan of `with_args_callback` either.  I will think about it more too.